src/version_map.rs

```diff
@@ -62,22 +62,54 @@
 
 use std::any::TypeId;
 use std::collections::hash_map::HashMap;
+use std::fmt::Debug;
+use std::sync::Arc;
 
 const BASE_VERSION: u16 = 1;
 
+/// Trait to check whether is specific `version` is supported by a `VersionMap`.
+pub trait VersionFilter: Debug {
```

Comment:
Why need to impl debug here?

Reply:
VersionMap's `[derive(Debug)]` depends on it.